### DISCUSSION TITLE
Handlebar if not working

### DISCUSSION THREAD

**Discussion by Oegl** [microsoft-search/pnp-modern-search#1629] [Q&A] [answered]
**Version used**
PnP Modern Search Webparts - v4 (Vers. 4.4.1.0)

**Describe the bug**
Within PnP Search results webpart, I use a slider template and add custom values => ends up in a custom result template. In this template I added the following
{{#if eoShowInterviewLinkOWSBOOL }}
<div class="eoInterviewArticleURL"><a href="\{{eoInterviewArticleURLOWSTEXT}}" target="_blank" data-interception="off">Zum Interview</a></div>
{{/if}}

"eoShowInterviewLinkOWSBOOL" is a managed property for a boolean field with the SPO list.

In Debug template I see that the values for "eoShowInterviewLinkOWSBOOL" are set correctly:
![image](https://user-images.githubusercontent.com/65904352/145817198-a7c25641-5720-417b-87bb-73b476df3e88.png)

**To Reproduce**
1. Create new SiteColumn "eoShowInterviewLink" of Type "Yes/No"
2. Add sitecolumn to a list
3. Create new entries in that list
4. Check if ManagedProperty "eoShowInterviewLinkOWSBOOL" for the site column is created
5. Add PnP Search Result webpart to a page
6. In the search result webpart settings add the managed property "eoShowInterviewLinkOWSBOOL"
7. In the search result webpart setting for the result template, select "custom" and add the {{#if eoShowInterviewLinkOWSBOOL }} ...<only visible if eoShowInterviewLinkOWSBOOL is set to Yes (1) ... {{/if}} 

But the "If" never takes action, doesn't matter if the column is set to Yes or No.

I also tried to solve this via a slot entry:
![image](https://user-images.githubusercontent.com/65904352/145820411-b3b96007-0535-434b-8cb7-16c7b6cf92a1.png)

and modified my if statement to the following:
{{#if (slot item @root.slots.ShowInterviewArticleURL)}}
<div class="eoInterviewArticleURL"><a href="\{{eoInterviewArticleURLOWSTEXT}}" target="_blank" data-interception="off">Zum Interview</a></div>
{{/if}}

But this also not ended up with the expected result!

Maybe I'm doing wrong.

Thanks for you help.

**Comment by wobba** [comment, marked as answer]
You should look at using the `eq` operator instead as the value you get back is a string, not a Boolean value.

https://github.com/helpers/handlebars-helpers/blob/master/lib/comparison.js#L165

**Reply by Oegl** [reply]
Perfect - thanks for your hint!!!!

**Reply by Oegl** [reply]
Sorry, one more question: Should this also work in the "Slider" result template?

I tried to do the following, but it didn't work:
....
 <pnp-slider ...>
		
	
		{{#eq eoShowInterviewLinkOWSBOOL '1'}}
			<div class="eoInterviewArticleURL">link</div>
		{{/eq}}      
< /pnp-slider>
....

**Reply by Oegl** [reply]
Solved it, I also have to escape {{#eq}} and {{/eq}} with a leading "Backslash"
